apps/admin/frontend/src/screens/manual_data_import_precinct_screen.tsx

```diff
@@ -308,25 +302,81 @@ export function ManualDataImportPrecinctScreen(): JSX.Element {
   const currentPrecinct = election.precincts.find(
     (p) => p.id === currentPrecinctId
   );
-  const [currentPrecinctTally, setCurrentPrecinctTally] =
-    useState<TempManualTally>(
-      assertDefined(initialTalliesByPrecinct[currentPrecinctId])
-    );
+  const [currentPrecinctTally, setCurrentPrecinctTally] = useState<
+    Optional<TempManualTally>
+  >(initialTalliesByPrecinct[currentPrecinctId]);
+
+  const [tempWriteInCandidates, setTempWriteInCandidates] = useState<
+    TempWriteInCandidate[]
+  >([]);
 
   async function saveResults() {
-    // Convert the temporary data structure that allows empty strings or
-    // numbers for all tallies to fill in 0s for any empty strings.
-    const convertedCurrentPrecinctTally: ManualTally = {
+    assert(currentPrecinctTally);
+    // replace temporary tally values with the numeric values we'll save
+    const finalPrecinctTally: ManualTally = {
       ...currentPrecinctTally,
       contestTallies: convertContestTallies(
         currentPrecinctTally.contestTallies
       ),
     };
 
-    const manualTally = convertTalliesByPrecinctToFullManualTally(
+    // remove any temporary write-in candidates with 0 votes
+    const nonZeroTempWriteInCandidates: TempWriteInCandidate[] = [];
+    for (const writeInCandidate of tempWriteInCandidates) {
+      const numVotes =
+        finalPrecinctTally.contestTallies[writeInCandidate.contestId]?.tallies[
+          writeInCandidate.id
+        ]?.tally;
+
+      if (!numVotes) {
+        delete finalPrecinctTally.contestTallies[writeInCandidate.contestId]
+          ?.tallies[writeInCandidate.id];
+      } else {
+        nonZeroTempWriteInCandidates.push(writeInCandidate);
+      }
+    }
+    try {
+      // add temporary write-in candidates to backend, get ids
+      const writeInCandidateRecords = await Promise.all(
+        nonZeroTempWriteInCandidates.map((candidate) =>
+          addWriteInCandidateMutation.mutateAsync({
+            contestId: candidate.contestId,
+            name: candidate.name,
+          })
+        )
+      );
```

Comment:
It's a bit convoluted to do a bunch of requests here. It would be better to do it as one mutation, but this works and it is temporary and will be removed soon.

When manual tally data is moved to the backend, you'll just submit the whole "form" including the temporary write-in candidates and the backend will handle adding or removing candidates as necessary.